src/server/polaris-server.ts

```diff
@@ -115,6 +137,21 @@ export class PolarisServer {
         };
     }
 
+    private getSchemaWithMiddlewares(): GraphQLSchema {
+        const schema = makeExecutablePolarisSchema(
+            this.polarisServerConfig.typeDefs,
+            this.polarisServerConfig.resolvers,
+            this.polarisServerConfig.schemaDirectives,
+        );
+        const middlewares = this.getAllowedPolarisMiddlewares();
+        if (this.polarisServerConfig.customMiddlewares) {
+            middlewares.push(...this.polarisServerConfig.customMiddlewares);
+        }
+
+        applyMiddleware(schema, new SnapshotMiddleware().getMiddleware());
```

Comment:
push it also to middlewares object before and applyMiddleware once instead of twice.

Reply:
we need to apply it first, so it will be called before every other middleware